### ISSUE TITLE
'Back to top' link moves focus after the 'skip to main content link' but before the navigation

### ISSUE THREAD

**Issue by abbott567** [alphagov/govuk-design-system#1561]
Hi everyone. I forked the design system for our accessibility manual, and I have a query about the back-to-top link. 

I've copied the way it works in the design system, which is to give the pane wrapper the id of top and then use `#top` as the href. 
However, the sequence of elements in the layout is: 
``` html
<body>
  <skip-link>
  <#top>
    <header>
    <nav>
    <subnav>
    <main>
```
This means when you use the back to top link, you're now at the top of the page above all of the navigational elements, but you're still under the skip-link which jumps you to the `<main>` tag. It means you can no longer skip past the navs etc for WCAG [2.4.1 bypass blocks](https://www.w3.org/WAI/WCAG21/Understanding/bypass-blocks.html) unless you know to press `shift + tab` to go one link up from the element `#top`.

It feels like the back to top link should jump to the top of the body, this way the skip-link would still be accessible in a logical reading order.

**Comment by edwardhorsford** [comment]
@abbott567 I wonder if it should take you to `main` - basically the same as the skip link in function.

**Comment by abbott567** [comment]
@edwardhorsford Yeah, I think if it went back to the top of the main or back to the top of the body those would make sense. But back to "not quite the top" feels like it could be disorientating.

**Comment by 36degrees** [comment]
Thanks for raising, @abbott567. I've prototyped the two options suggested so far. I believe either would resolve this issue…

@christopherthomasdesign @CharlotteDowns any thoughts on which of these makes more sense?

## Moving the `#top` anchor to before the skip link

Activating the 'back to top' link takes the user to the very top of the page:

![Screenshot 2021-11-09 at 15 37 23](https://user-images.githubusercontent.com/121939/140955356-6ac17506-0954-4967-9215-836908da23a6.png)

The next <kbd>tab</kbd> takes the user to the 'skip to main content' link:

![Screenshot 2021-11-09 at 15 37 27](https://user-images.githubusercontent.com/121939/140955367-0467e776-3820-4084-bf25-3fae1b300b0e.png)

## Changing the target of the 'back to top' link to `#main-content`

This is the same place that the 'Skip to main content' link takes users to.

Activating the 'back to top' link takes the user to the start of the content, after the heading:

![Screenshot 2021-11-09 at 15 37 59](https://user-images.githubusercontent.com/121939/140955532-9b68c7f1-2579-4340-9cee-5a5871624518.png)

The next <kbd>tab</kbd> takes the user to the first interactive control within the guidance on that page:

![Screenshot 2021-11-09 at 15 35 13](https://user-images.githubusercontent.com/121939/140955605-9a13b755-7abb-42a0-a509-98d3d3c161dd.png)

**Comment by christopherthomasdesign** [comment]
I *think* option 2. IMO the context of a 'back to top' link is usually within the main body of a page, as opposed to before the nav etc. I know it's all one _web_ page, but persistent elements like navigation maybe sit outside of how I'd think about a page. That's just my interpretation though, I don't fully understand the impact each would have on users of different assistive tech.

**Comment by jbuller** [comment]
Going to #main feels sensible to me, as going to the top of the page can be achieved in other ways.

In some cases 'Back to contents list' or similar might be more appropriate.
Maybe if such a link Is deemed necessary it implies a contents list is needed too?

If you'd been linked into the middle of a page though it wouldn't be going 'Back' so maybe '[Up to] List of Contents' ?

**Comment by abbott567** [comment]
I think @jbuller is onto something there. Back to top is a bit subjective. 

I took it literally like "back to top" was going to the top of the rendered page which would be the start of the `<body>`, which is why it seemed weird that it was under the skip link. However, I do acknowledge that my brain is overly technical a lot of the time so it's unlikely this would be a common thought process.

I think either of the suggestions by @36degrees  would work better than the current solution. Though I do think the second option could be slightly more frustrating if you're actually trying to get back to the navigation menu from midway through the page. 

If you're using a screen reader you could jump between landmarks using hot keys, but if you're just using a keyboard you're kinda just left with the tab key. Maybe I'm missing something, but the only options I could see to get back to the nav would be:
- use `TAB` to go all the way through the content and footer content until you eventually get back to the top of the page
- use back to top to get to the top of the main, and then use `SHIFT+TAB` to reverse through the nav menu
- refresh the page and use `TAB`

**Comment by jbuller** [comment]
[NVDA Ctrl+] Home too but I guess not on all (mobile) keyboards

**Comment by jbuller** [comment]
More on the same topic https://www.digitala11y.com/scroll-to-top-where-should-the-focus-land/

**Comment by 36degrees** [comment]
I've been digging in to the history of the back to top link to try and understand what we actually intended it to do.

The back to top link was introduced in #687, when we moved from the side navigation and the main content being fixed, independently scrollable panes. This change meant that the side navigation no longer remained within the viewport.

The description for that PR says:

> We have created a new back to top component that **will return you to the navigation**, since some of our pages are long and no longer have the navigation available as you scroll.

So, it seems like the intent at the time was that the back to top link would bring the side navigation back into view, rather than providing a convenient way for the user to get back to the top of the main content.

It's hard for us to know whether our intent matches with a user's expectation of what the back to top link should behave, or what it's for, but based on this I _think_ it makes more sense to go with 'option 1' and return the user to the very top of the page. That seems to provide the 'best of both worlds' as the user can still choose to bypass the navigation by using the skip link, if they were aiming to get back to the top of the content.

This seems to fit with @abbott567's thoughts about the second option potentially being more frustrating for users aiming to get back to the navigation.

Thoughts?

**Comment by jbuller** [comment]
Salient point from a discussion on an accessibility forum:
> shortcut keys like Home or Ctrl+Home, while they do scroll the page to the top, they don't move your focus to the top.  That is, if my keyboard focus is in the middle of the page and I hit Home and then Tab, my focus won't start at the top of the page.
